### PR TITLE
MODCLUSTER-790 mod_cluster listener throws NPE when deploying a context in Tomcat - refine ContainerEventHandler#add contract and implementation

### PR DESCRIPTION
**Pull request by rhusar**
Resolves
https://issues.redhat.com/browse/MODCLUSTER-790

### CODE REVIEW

File: container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java [hunk 1]
```diff
@@ -169,7 +169,10 @@ public void deployWebApp() throws Exception {
 
         verify(context).addLifecycleListener(handler);
         verify(context).addPropertyChangeListener(handler);
-        verify(this.eventHandler).add(eq(catalinaContext));
+
+        LifecycleEvent lifecycleEvent = new LifecycleEvent(context, Lifecycle.AFTER_START_EVENT,context);
```
Comment (pferraro) [inline container/tomcat-10.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:173]: Cosmetic formatting issue.

File: container/tomcat-8.5/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java [hunk 2]
```diff
@@ -169,7 +169,10 @@ public void deployWebApp() throws Exception {
 
         verify(context).addLifecycleListener(handler);
         verify(context).addPropertyChangeListener(handler);
-        verify(this.eventHandler).add(eq(catalinaContext));
+
+        LifecycleEvent lifecycleEvent = new LifecycleEvent(context, Lifecycle.AFTER_START_EVENT,context);
```
Comment (pferraro) [inline container/tomcat-8.5/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:173]: Cosmetic formatting issue.

File: container/tomcat-9.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java [hunk 3]
```diff
@@ -169,7 +169,10 @@ public void deployWebApp() throws Exception {
 
         verify(context).addLifecycleListener(handler);
         verify(context).addPropertyChangeListener(handler);
-        verify(this.eventHandler).add(eq(catalinaContext));
+
+        LifecycleEvent lifecycleEvent = new LifecycleEvent(context, Lifecycle.AFTER_START_EVENT,context);
```
Comment (pferraro) [inline container/tomcat-9.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java:173]: Cosmetic formatting issue.

**Review by pferraro — CHANGES_REQUESTED** [review]
Logic looks good.

File: container/tomcat-9.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java (hunk 3 above)
Comment (rhusar) [reply container/tomcat-9.0/src/test/java/org/jboss/modcluster/container/tomcat/ContainerEventHandlerAdapterTestCase.java]: Oops, thanks.

**Pull request merged**